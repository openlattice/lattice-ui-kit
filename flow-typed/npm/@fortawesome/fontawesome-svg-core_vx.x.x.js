// flow-typed signature: 39887e5e6c6c32861704232d9719cd14
// flow-typed version: <<STUB>>/@fortawesome/fontawesome-svg-core_v1.2.4/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/fontawesome-svg-core'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/fontawesome-svg-core' {
  declare module.exports: any;
}
