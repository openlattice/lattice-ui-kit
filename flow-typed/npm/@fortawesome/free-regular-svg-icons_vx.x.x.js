// flow-typed signature: e06b1e176fa03c6cedb9f03bb6cc7872
// flow-typed version: <<STUB>>/@fortawesome/free-regular-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-regular-svg-icons' {
  declare module.exports: any;
}
