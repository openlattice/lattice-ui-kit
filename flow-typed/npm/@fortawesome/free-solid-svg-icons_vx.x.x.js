// flow-typed signature: 64f61153e222e791ad6eca6c60c55baa
// flow-typed version: <<STUB>>/@fortawesome/free-solid-svg-icons_v5.13.0/flow_v0.118.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/free-solid-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/free-solid-svg-icons' {
  declare module.exports: any;
}
