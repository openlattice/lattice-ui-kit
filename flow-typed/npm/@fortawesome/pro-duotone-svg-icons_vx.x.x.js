// flow-typed signature: dc91210f6aed06e504e8c5d05af6775b
// flow-typed version: <<STUB>>/@fortawesome/pro-duotone-svg-icons_v5.10.2/flow_v0.106.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-duotone-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-duotone-svg-icons' {
  declare module.exports: any;
}
