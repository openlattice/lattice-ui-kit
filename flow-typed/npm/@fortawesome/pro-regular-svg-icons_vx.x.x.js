// flow-typed signature: c41877c2f635a0484c823342d488e66e
// flow-typed version: <<STUB>>/@fortawesome/pro-regular-svg-icons_v5.3.1/flow_v0.80.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@fortawesome/pro-regular-svg-icons'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@fortawesome/pro-regular-svg-icons' {
  declare module.exports: any;
}
