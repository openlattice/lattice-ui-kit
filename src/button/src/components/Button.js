/*
 * @flow
 */

import React, { Component } from 'react';
import type { Node } from 'react';

import PropTypes from 'prop-types';

import DefaultButton from './styled/DefaultButton';
import PrimaryButton from './styled/PrimaryButton';
import SecondaryButton from './styled/SecondaryButton';

type ButtonMode =
  | 'default'
  | 'primary'
  | 'secondary';

type Props = {
  children :Node;
  className ? :string;
  disabled ? :boolean;
  mode ? :ButtonMode;
  onClick :(e :SyntheticEvent<HTMLButtonElement>) => void;
  type ? :string;
};

type State = {};

/*
 * Inspiration:
 * https://atlaskit.atlassian.com/packages/core/button
 * https://evergreen.surge.sh/components/buttons
 */
export default class Button extends Component<Props, State> {

  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    mode: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    type: PropTypes.string,
  }

  static defaultProps = {
    className: '',
    disabled: false,
    mode: 'default',
    type: 'button',
  }

  render() {

    const {
      children,
      className,
      disabled,
      mode,
      onClick,
      type,
    } = this.props;

    switch (mode) {
      case 'primary':
        return (
          <PrimaryButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </PrimaryButton>
        );
      case 'secondary':
        return (
          <SecondaryButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </SecondaryButton>
        );
      default:
        return (
          <DefaultButton className={className} disabled={disabled} onClick={onClick} type={type}>
            { children }
          </DefaultButton>
        );
    }
  }
}
