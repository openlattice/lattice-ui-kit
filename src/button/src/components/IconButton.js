/*
 * @flow
 */

import React, { useMemo } from 'react';

import clsx from 'clsx';
import styled from 'styled-components';
import { IconButton as MuiIconButton } from '@material-ui/core';

import useButtonStyles, { isCustomColor } from './useButtonStyles';
import { styleName } from './createColorStyles';
import type { ButtonProps } from './types';

import Spinner from '../../../spinner';

const IconSquareWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  width: auto;

  &::before {
    content: "";
    display: block;
    padding-bottom: 100%; /* padding size is relative to the width */
  }
`;

const SpinnerWrapper = styled.div`
  align-items: center;
  background: white;
  border-radius: 50%;
  display: flex;
  height: 100%;
  justify-content: center;
  position: absolute;
  width: 100%;
`;

const IconButton = ({
  children,
  className,
  color = 'default',
  disabled = false,
  isLoading = false,
  ...rest
} :ButtonProps) => {

  const classes = useButtonStyles();
  const customColor = useMemo(() => isCustomColor(color), [color]);

  // https://medium.com/flow-type/spreads-common-errors-fixes-9701012e9d58
  // [...] you can annotate the object with optional properties and then explicitly set the property afterwards
  const classesObject = {
    [classes[styleName('text', color)]]: customColor,
  };
  if (className) {
    classesObject[className] = className;
  }

  const props = {
    ...rest,
    className: clsx(classesObject),
    color: customColor ? 'default' : color,
    disabled: disabled || isLoading,
  };

  /* eslint-disable react/jsx-props-no-spreading */
  return (
    <MuiIconButton {...props}>
      <IconSquareWrapper>
        {
          isLoading && (
            <SpinnerWrapper>
              <Spinner />
            </SpinnerWrapper>
          )
        }
        {children}
      </IconSquareWrapper>
    </MuiIconButton>
  );
  /* eslint-enable */
};

export default IconButton;
