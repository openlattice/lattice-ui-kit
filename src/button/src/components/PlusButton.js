/*
 * @flow
 */

import React from 'react';

import { faPlus } from '@fortawesome/pro-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import IconButton from './IconButton';
import type { Props } from './Button';

const icon = (
  <FontAwesomeIcon icon={faPlus} />
);

/* eslint-disable react/jsx-props-no-spreading */
const PlusButton = (props :Props) => (
  <IconButton icon={icon} {...props} />
);
/* eslint-enable */

export default PlusButton;
