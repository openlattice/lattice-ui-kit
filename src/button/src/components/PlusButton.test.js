import React from 'react';
import toJson from 'enzyme-to-json';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { mount } from 'enzyme';

import Button from './Button';
import IconButton from './IconButton';
import PlusButton from './PlusButton';

const BUTTON_TXT = 'Edit';

describe('PlusButton', () => {

  describe('icon with text', () => {

    const plusButton = mount(
      <PlusButton>{BUTTON_TXT}</PlusButton>
    );

    test('should match snapshot', () => {
      expect(toJson(plusButton)).toMatchSnapshot();
    });

    test('should render a button element', () => {
      expect(plusButton.find('button')).toHaveLength(1);
    });

    test('should render FontAwesomeIcon component', () => {
      expect(plusButton.find(FontAwesomeIcon)).toHaveLength(1);
    });

    test('should render IconButton component', () => {
      expect(plusButton.find(IconButton)).toHaveLength(1);
    });

    test('should render Button component', () => {
      expect(plusButton.find(Button)).toHaveLength(1);
    });

    test('should render the correct text', () => {
      expect(plusButton.text()).toEqual(BUTTON_TXT);
    });

  });

  describe('icon without text', () => {

    const plusButton = mount(
      <PlusButton />
    );

    test('should match snapshot', () => {
      expect(toJson(plusButton)).toMatchSnapshot();
    });

    test('should render a button element', () => {
      expect(plusButton.find('button')).toHaveLength(1);
    });

    test('should render FontAwesomeIcon component', () => {
      expect(plusButton.find(FontAwesomeIcon)).toHaveLength(1);
    });

    test('should render IconButton component', () => {
      expect(plusButton.find(IconButton)).toHaveLength(1);
    });

    test('should render Button component', () => {
      expect(plusButton.find(Button)).toHaveLength(1);
    });

    test('should not render any text', () => {
      expect(plusButton.text()).toEqual('');
    });

  });

});
