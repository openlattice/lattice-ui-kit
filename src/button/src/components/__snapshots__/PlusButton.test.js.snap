// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusButton icon with text should match snapshot 1`] = `
.c0 {
  margin: 0 8px 0 0;
}

<PlusButton>
  <IconButton
    icon={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              384,
              512,
              Array [],
              "f067",
              "M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z",
            ],
            "iconName": "plus",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
  >
    <Button>
      <WithStyles(ForwardRef(Button))
        className=""
        color="default"
        disableElevation={true}
        variant="contained"
      >
        <ForwardRef(Button)
          className=""
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit",
              "contained": "MuiButton-contained",
              "containedPrimary": "MuiButton-containedPrimary",
              "containedSecondary": "MuiButton-containedSecondary",
              "containedSizeLarge": "MuiButton-containedSizeLarge",
              "containedSizeSmall": "MuiButton-containedSizeSmall",
              "disableElevation": "MuiButton-disableElevation",
              "disabled": "Mui-disabled",
              "endIcon": "MuiButton-endIcon",
              "focusVisible": "Mui-focusVisible",
              "fullWidth": "MuiButton-fullWidth",
              "iconSizeLarge": "MuiButton-iconSizeLarge",
              "iconSizeMedium": "MuiButton-iconSizeMedium",
              "iconSizeSmall": "MuiButton-iconSizeSmall",
              "label": "MuiButton-label",
              "outlined": "MuiButton-outlined",
              "outlinedPrimary": "MuiButton-outlinedPrimary",
              "outlinedSecondary": "MuiButton-outlinedSecondary",
              "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
              "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
              "root": "MuiButton-root",
              "sizeLarge": "MuiButton-sizeLarge",
              "sizeSmall": "MuiButton-sizeSmall",
              "startIcon": "MuiButton-startIcon",
              "text": "MuiButton-text",
              "textPrimary": "MuiButton-textPrimary",
              "textSecondary": "MuiButton-textSecondary",
              "textSizeLarge": "MuiButton-textSizeLarge",
              "textSizeSmall": "MuiButton-textSizeSmall",
            }
          }
          color="default"
          disableElevation={true}
          variant="contained"
        >
          <WithStyles(ForwardRef(ButtonBase))
            className="MuiButton-root MuiButton-contained MuiButton-disableElevation"
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            type="button"
          >
            <ForwardRef(ButtonBase)
              className="MuiButton-root MuiButton-contained MuiButton-disableElevation"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              component="button"
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="Mui-focusVisible"
              type="button"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-disableElevation"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <IconButton__IconMarginRight>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "IconButton__IconMarginRight-g9q5gl-0",
                            "isStatic": false,
                            "lastClassName": "c0",
                            "rules": Array [
                              "margin:0 8px 0 0;",
                            ],
                          },
                          "displayName": "IconButton__IconMarginRight",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "IconButton__IconMarginRight-g9q5gl-0",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="c0"
                      >
                        <FontAwesomeIcon
                          border={false}
                          className=""
                          fixedWidth={false}
                          flip={null}
                          icon={
                            Object {
                              "icon": Array [
                                384,
                                512,
                                Array [],
                                "f067",
                                "M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z",
                              ],
                              "iconName": "plus",
                              "prefix": "far",
                            }
                          }
                          inverse={false}
                          listItem={false}
                          mask={null}
                          pull={null}
                          pulse={false}
                          rotation={null}
                          size={null}
                          spin={false}
                          swapOpacity={false}
                          symbol={false}
                          title=""
                          transform={null}
                        >
                          <svg
                            aria-hidden="true"
                            className="svg-inline--fa fa-plus fa-w-12 "
                            data-icon="plus"
                            data-prefix="far"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 384 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                              fill="currentColor"
                              style={Object {}}
                            />
                          </svg>
                        </FontAwesomeIcon>
                      </span>
                    </StyledComponent>
                  </IconButton__IconMarginRight>
                  Add
                </span>
                <WithStyles(memo)
                  center={false}
                >
                  <ForwardRef(TouchRipple)
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child",
                        "childLeaving": "MuiTouchRipple-childLeaving",
                        "childPulsate": "MuiTouchRipple-childPulsate",
                        "ripple": "MuiTouchRipple-ripple",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                        "root": "MuiTouchRipple-root",
                      }
                    }
                  >
                    <span
                      className="MuiTouchRipple-root"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component={null}
                        exit={true}
                      />
                    </span>
                  </ForwardRef(TouchRipple)>
                </WithStyles(memo)>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(Button)>
      </WithStyles(ForwardRef(Button))>
    </Button>
  </IconButton>
</PlusButton>
`;

exports[`PlusButton icon without text should match snapshot 1`] = `
<PlusButton>
  <IconButton
    icon={
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              384,
              512,
              Array [],
              "f067",
              "M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z",
            ],
            "iconName": "plus",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    }
  >
    <Button>
      <WithStyles(ForwardRef(Button))
        className=""
        color="default"
        disableElevation={true}
        variant="contained"
      >
        <ForwardRef(Button)
          className=""
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit",
              "contained": "MuiButton-contained",
              "containedPrimary": "MuiButton-containedPrimary",
              "containedSecondary": "MuiButton-containedSecondary",
              "containedSizeLarge": "MuiButton-containedSizeLarge",
              "containedSizeSmall": "MuiButton-containedSizeSmall",
              "disableElevation": "MuiButton-disableElevation",
              "disabled": "Mui-disabled",
              "endIcon": "MuiButton-endIcon",
              "focusVisible": "Mui-focusVisible",
              "fullWidth": "MuiButton-fullWidth",
              "iconSizeLarge": "MuiButton-iconSizeLarge",
              "iconSizeMedium": "MuiButton-iconSizeMedium",
              "iconSizeSmall": "MuiButton-iconSizeSmall",
              "label": "MuiButton-label",
              "outlined": "MuiButton-outlined",
              "outlinedPrimary": "MuiButton-outlinedPrimary",
              "outlinedSecondary": "MuiButton-outlinedSecondary",
              "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
              "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
              "root": "MuiButton-root",
              "sizeLarge": "MuiButton-sizeLarge",
              "sizeSmall": "MuiButton-sizeSmall",
              "startIcon": "MuiButton-startIcon",
              "text": "MuiButton-text",
              "textPrimary": "MuiButton-textPrimary",
              "textSecondary": "MuiButton-textSecondary",
              "textSizeLarge": "MuiButton-textSizeLarge",
              "textSizeSmall": "MuiButton-textSizeSmall",
            }
          }
          color="default"
          disableElevation={true}
          variant="contained"
        >
          <WithStyles(ForwardRef(ButtonBase))
            className="MuiButton-root MuiButton-contained MuiButton-disableElevation"
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            type="button"
          >
            <ForwardRef(ButtonBase)
              className="MuiButton-root MuiButton-contained MuiButton-disableElevation"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              component="button"
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="Mui-focusVisible"
              type="button"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-disableElevation"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          384,
                          512,
                          Array [],
                          "f067",
                          "M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z",
                        ],
                        "iconName": "plus",
                        "prefix": "far",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-plus fa-w-12 "
                      data-icon="plus"
                      data-prefix="far"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 384 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </span>
                <WithStyles(memo)
                  center={false}
                >
                  <ForwardRef(TouchRipple)
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child",
                        "childLeaving": "MuiTouchRipple-childLeaving",
                        "childPulsate": "MuiTouchRipple-childPulsate",
                        "ripple": "MuiTouchRipple-ripple",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                        "root": "MuiTouchRipple-root",
                      }
                    }
                  >
                    <span
                      className="MuiTouchRipple-root"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component={null}
                        exit={true}
                      />
                    </span>
                  </ForwardRef(TouchRipple)>
                </WithStyles(memo)>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(Button)>
      </WithStyles(ForwardRef(Button))>
    </Button>
  </IconButton>
</PlusButton>
`;
